src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -219,9 +224,57 @@ public Task PublishUpdateAsync(IResource resource, string resourceId, Func<Custo
                     newState.ExitCode, string.Join(", ", newState.EnvironmentVariables.Select(e => $"{e.Name} = {e.Value}")), string.Join(", ", newState.Urls.Select(u => $"{u.Name} = {u.Url}")),
                     string.Join(", ", newState.Properties.Select(p => $"{p.Name} = {p.Value}")));
             }
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static ImmutableArray<ResourceCommandSnapshot> BuildCommands(IResource resource, CustomResourceSnapshot previousState)
```

Comment:
This is too expensive to do every publish. The goal is to figure out which commands are visible given the state correct?

Reply:
I'll look at making it faster, but how often are you expecting resources to be published? It's not like we need to do thousands of updates per second.

Reply:
every 5 seconds after the health checks PR is made more aggressive.